Lib/test/test_codeop.py

```diff
@@ -294,6 +294,11 @@ def test_filename(self):
         self.assertNotEqual(compile_command("a = 1\n", "abc").co_filename,
                             compile("a = 1\n", "def", 'single').co_filename)
 
+    def test_warning(self):
+        # Test that the warning is only returned once.
+        with support.check_warnings((".*invalid", DeprecationWarning)) as w:
+            compile_command("\e")
```

Comment:
You forgot that the outer quotes are not part of the compiled syntax.  (I have done this too.)  `\e` is a syntax error, `'\e'` is valid and gives no warning.

Reply:
Sorry about that.  I was actually having trouble getting the '\e' to show the Deprecation Warning on any release until I started the REPL or IDLE with the `./python -Wall`.  I did this change quickly just to see how the tests would be run and it looks like I did it too quickly.  :-(

Reply:
The important point is `'\e'` warns when starting IDLE (legitimately) with `py -3.7 -Wall -m idlelib`.  So the backport is needed.

Should the tests always pass, not just with CI?  At least if one does not explicitly turn off warnings for tests (which seems like a bad idea)?  I made a local 3.7 branch from this PR and ran `python -m test test_codeop` and it passed normally.  `python -m test.test_codeop` reports 5 tests rather than the previous 4 and prints the DeprecationWarning.

So I will merge and we will see how this fares on buildbots.

Reply:
20 seconds later I realize that warnings <= 1 is sufficient to prove not 3 or even 2.  But 0 could also result from a bad patch that suppressed too much.  So if a fix is needed, better to add a context forcing the warning on.